ci-operator/step-registry/gather/alibabacloud-console/gather-alibabacloud-console-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export ALIBABA_CLOUD_CREDENTIALS_FILE=${SHARED_DIR}/inicreds
+export ALIBABA_CLI_CREDENTIALS_FILE=${SHARED_DIR}/config
+
+wget https://aliyuncli.alicdn.com/aliyun-cli-linux-latest-amd64.tgz -O /tmp/aliyun-cli.tgz
+pushd /tmp
+tar zxvf aliyun-cli.tgz
+popd
+
+REGION="${LEASED_RESOURCE}"
+/tmp/aliyun version
+
+echo "Settting --config-path=${ALIBABA_CLI_CREDENTIALS_FILE} and --region=${LEASED_RESOURCE}"
+/tmp/aliyun  --config-path ${ALIBABA_CLI_CREDENTIALS_FILE} configure set --region "${LEASED_RESOURCE}"
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+	exit 0
```

Comment:
nit: seems like if/else block could be shifted up.  No need to pull binaries or set regions or anything if we don't have a `KUBECONFIG`.

Reply:
I think we add `alibaba-instance-ids.txt` by snarfing the .openshift-install.log. I will move the download for the binary down in the other `test -f` section.